apps/api/app/utils/notification/channels/telegram.py

```
@@ -0,0 +1,211 @@
+        self, session: aiohttp.ClientSession, ctx: Dict[str, Any]
+    ) -> tuple[SendFn | None, ChannelDeliveryStatus | None]:
+        """Store (session, ctx) as instance vars so _deliver_content can reach them.
+
+        Returns a no-op SendFn to satisfy the base class contract; the actual
+        sending is done via _send_message in our _deliver_content override.
+        """
+        # Stash for _deliver_content (cleaned up in finally block below).
+        self._tg_session = session  # type: ignore[attr-defined]
+        self._tg_ctx = ctx  # type: ignore[attr-defined]
+
+        async def _noop(text: str) -> str | None:  # satisfies SendFn signature
```

Ruff (0.15.1)

[warning] Unused function argument: `text`

(ARG001)